cmd/flags/auth.go

```diff
@@ -108,26 +154,40 @@ Environment variable: %q`, authOIDCCACertFileEnv)
 }
 
 func NewAuthConfig(vipr *viper.Viper) (*auth.Config, error) {
-	certs := []*x509.Certificate{}
+	// OIDC part
+	certsOIDC := []*x509.Certificate{}
 
-	fileCert, err := fileCertificate(vipr)
+	fileCertOIDC, err := fileCertificate(vipr)
 	if err != nil {
 		return nil, err
-	} else if fileCert != nil {
-		certs = append(certs, fileCert)
+	} else if fileCertOIDC != nil {
+		certsOIDC = append(certsOIDC, fileCertOIDC)
 	}
 
 	issuerCerts, err := issuerCertificates(vipr)
 	if err != nil {
 		return nil, err
 	} else if issuerCerts != nil {
-		certs = append(certs, issuerCerts...)
+		certsOIDC = append(certsOIDC, issuerCerts...)
 	}
 
 	oidcCfg := oidc.NewConfig(vipr.GetString(authOIDCClaimUsernameViperKey),
-		vipr.GetString(authOIDCClaimGroupViperKey), certs...)
+		vipr.GetString(authOIDCClaimGroupViperKey), certsOIDC...)
+
+	// TLS part
+	certsTLS := []*x509.Certificate{}
+
+	fileCertTLS, err := tlsAuthClientCertificate(vipr)
+	if err != nil {
+		return nil, err
+	} else if fileCertTLS != nil {
+		certsTLS = append(certsTLS, fileCertTLS)
```

Comment:
Why you need to append into an array instead of doing the assigment at the initialization?

Reply:
Well I guess I just did copy/paste the above code. Better without indeed.